caddyconfig/httpcaddyfile/tlsapp.go

```diff
@@ -497,7 +497,7 @@ func consolidateAutomationPolicies(aps []*caddytls.AutomationPolicy) []*caddytls
 			// if they're exactly equal in every way, just keep one of them
 			if reflect.DeepEqual(aps[i], aps[j]) {
 				aps = append(aps[:j], aps[j+1:]...)
-				i--
+				i = max(0, i-1)
```

Comment:
I'm actually wondering if decrementing `i` here is the bug, not decrementing it past 0... hmm.